bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -156,8 +156,8 @@ public ShellyBaseHandler(final Thing thing, final ShellyTranslationProvider tran
         Map<String, String> properties = thing.getProperties();
         String gen = getString(properties.get(PROPERTY_DEV_GEN));
         String thingType = getThingType();
-        gen2 = !"1".equals(gen) || ShellyDeviceProfile.isGeneration2(thingType);
         blu = ShellyDeviceProfile.isBluSeries(thingType);
+        gen2 = "2".equals(gen) || "3".equals(gen) || blu || ShellyDeviceProfile.isGeneration2(thingType);
```

Comment:
Can thing type be ambiguous, so that you can't determine generation from it?

If it can, I guess this will still be broken for users having manually created things?

It it cannot, why check the property at all to determine generation? Would this work:
```java
gen2 = blu || ShellyDeviceProfile.isGeneration2(thingType);
```

?

Reply:
That is exactly one of my questions in previous @J-N-K PR.

Reply:
It depends on whether this property is only informative but not required.

Reply:
I know, but since the property is still checked in this PR, I wanted to ask it again and even more explicit to fully understand. I can see only one reason for checking the property at all, and that would be some corner case where thing type is not sufficient. In that case, even this fix would not work.

Reply:
Yes, you're right.

Reply:
Here are the thing type-based generation checks:
https://github.com/openhab/openhab-addons/blob/ab385ed2f35612cc447dbe992896d6c0e0c487fb/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java#L406-L413

I tried to apply this logic the all thing types:
| Thing Type ID        | Label                   | Result |
| -------------------- | ----------------------- | ------ |
| shellyblubutton      | Shelly BLU Button       | blu    |
| shellybludw          | Shelly BLU Door/Window  | blu    |
| shellyblumotion      | Shelly BLU Motion       | blu    |
| shellybluht          | Shelly BLU H&T          | blu    |
| shellyblugw          | Shelly BLU Gateway      | gen1   |
| shellybulb           | Shelly Bulb             | gen1   |
| shellybulbduo        | Shelly Duo              | gen1   |
| shellycolorbulb      | Shelly Color Bulb       | gen1   |
| shellyvintage        | Shelly Vintage          | gen1   |
| shellyrgbw2-color    | Shelly RGBW2 Color Mode | gen1   |
| shellyrgbw2-white    | Shelly RGBW2 White Mode | gen1   |
| shelly1              | Shelly 1                | gen1   |
| shelly1l             | Shelly 1L               | gen1   |
| shelly1pm            | Shelly 1PM              | gen1   |
| shellyem             | Shelly EM               | gen1   |
| shellyem3            | Shelly EM3              | gen1   |
| shelly2-relay        | Shelly 2 Relay          | gen1   |
| shelly2-roller       | Shelly 2 Roller         | gen1   |
| shelly25-relay       | Shelly 2.5 Relay        | gen1   |
| shelly25-roller      | Shelly 2.5 Roller       | gen1   |
| shelly4pro           | Shelly 4 Pro Relay      | gen1   |
| shellyplug           | Shelly Plug             | gen1   |
| shellyplugs          | Shelly Plug-S           | gen1   |
| shellyplugu1         | Shelly Plug US          | gen1   |
| shellyuni            | Shelly UNI              | gen1   |
| shellydimmer         | Shelly Dimmer           | gen1   |
| shellydimmer2        | Shelly Dimmer 2         | gen1   |
| shellyix3            | Shelly ix3              | gen1   |
| shellyht             | Shelly H&T              | gen1   |
| shellysmoke          | Shelly Smoke            | gen1   |
| shellygas            | Shelly Gas              | gen1   |
| shellyflood          | Shelly Flood            | gen1   |
| shellydw             | Shelly Door/Window      | gen1   |
| shellydw2            | Shelly Door/Window      | gen1   |
| shellysense          | Shelly Sense            | gen1   |
| shellybutton1        | Shelly Button 1         | gen1   |
| shellybutton2        | Shelly Button 2         | gen1   |
| shellymotion         | Shelly Motion           | gen1   |
| shellytrv            | Shelly TRV              | gen1   |
| shellyplus1          | ShellyPlus 1            | gen2   |
| shellyplus1pm        | ShellyPlus 1PM          | gen2   |
| shellyplus2pm-relay  | ShellyPlus 2 Relay      | gen2   |
| shellyplus2pm-roller | ShellyPlus 2PM Roller   | gen2   |
| shellyplusplug       | ShellyPlus Plug         | gen2   |
| shellyplusi4         | ShellyPlus i4           | gen2   |
| shellyplusi4dc       | ShellyPlus i4DC         | gen2   |
| shelly1mini          | ShellyPlus 1 Mini       | gen2   |
| shellypmmini         | ShellyPlus PM Mini      | gen2   |
| shelly1pmmini        | ShellyPlus 1PM Mini     | gen2   |
| shellypro1           | ShellyPro 1             | gen2   |
| shellypro1pm         | ShellyPro 1PM           | gen2   |
| shellypro2-relay     | ShellyPro 2 Relay       | gen2   |
| shellypro2pm-relay   | ShellyPro 2PM Relay     | gen2   |
| shellypro2pm-roller  | ShellyPro 2PM Roller    | gen2   |
| shellypro3           | ShellyPro 3             | gen2   |
| shellypro3em         | Shelly Pro 3EM          | gen2   |
| shellyproem50        | Shelly Pro EM-50        | gen2   |
| shellypro4pm         | ShellyPro 4PM           | gen2   |
| shellypluswdus       | Shelly Plus Dimmer US   | gen2   |
| shellyplus10v        | Shelly Plus Dimmer 10V  | gen2   |
| shellyhtg3           | ShellyPlus H&T Gen 3    | gen2   |
| shellyplussmoke      | Shelly Plus Smoke       | gen2   |
| shellywalldisplay    | Shelly Wall Display     | gen1   |
| shellydevice         | Shelly Device           | gen1   |
| shellyunknown        | Unknown Shelly Device   | gen1   |

Does it look good to you?